### PR TITLE
test: fix tests on windows

### PR DESCRIPTION
**Pull request by viceice**
## Changes
fix test spy reset

## Context
- closes #20411

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by JamieMagee — APPROVED** [review]

**Review by secustor — APPROVED** [review]

**Comment by viceice** [comment]
let's wait for windows tests to succeed before merge / automerge

**Review by rarkins — APPROVED** [review]

**Pull request merged**
